hybrasyl/Objects/Monster.cs

```diff
@@ -119,9 +231,282 @@ public override void Attack(Castables.Castable castObject, Creature target)
 
         public override void Attack(Castable castObject)
         {
-            //do monster aoe
+            var damage = castObject.Effects.Damage;
+            if (damage != null)
+            {
+                var intents = castObject.Intents;
+                foreach (var intent in intents)
+                {
+                    var possibleTargets = new List<VisibleObject>();
+                    Rectangle rect = new Rectangle(0, 0, 0, 0);
+
+                    switch (intent.Direction)
+                    {
+                        case IntentDirection.Front:
+                            {
+                                switch (Direction)
+                                {
+                                    case Direction.North:
+                                        {
+                                            //facing north, attack north
+                                            rect = new Rectangle(this.X, this.Y - intent.Radius, 1, intent.Radius);
+                                        }
+                                        break;
+                                    case Direction.South:
+                                        {
+                                            //facing south, attack south
+                                            rect = new Rectangle(this.X, this.Y, 1, 1 + intent.Radius);
+                                        }
+                                        break;
+                                    case Direction.East:
+                                        {
+                                            //facing east, attack east
+                                            rect = new Rectangle(this.X, this.Y, 1 + intent.Radius, 1);
+                                        }
+                                        break;
+                                    case Direction.West:
+                                        {
+                                            //facing west, attack west
+                                            rect = new Rectangle(this.X - intent.Radius, this.Y, intent.Radius, 1);
+                                        }
+                                        break;
```

Comment:
This looks like it is defining, effectively, an AOI / viewport to be used to select a target? I believe all of this should be refactored to use something like `GetAttackViewport_North/South/etc` with a radius parameter. There is an awful lot of repetition here. Many of these cases are effectively identical.

Reply:
Effectively Identical, but not identical.  This, again, was code copied from the user method and modified for monsters.

Reply:
Regardless, most of the viewport calculations are redundant and should be made less so.

Reply:
I'm not going to start refactoring out parts of methods into common methods unless it 100% warrants it. In this case, it does not. It's just a waste of time.